storage/client/src/main/java/com/azure/storage/queue/QueueServiceAsyncClient.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.storage.queue;
+
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.http.rest.VoidResponse;
+import com.azure.core.util.Context;
+import com.azure.storage.queue.implementation.AzureQueueStorageImpl;
+import com.azure.storage.queue.models.ListQueuesIncludeType;
+import com.azure.storage.queue.models.QueueItem;
+import com.azure.storage.queue.models.QueuesSegmentOptions;
+import com.azure.storage.queue.models.StorageErrorCode;
+import com.azure.storage.queue.models.StorageErrorException;
+import com.azure.storage.queue.models.StorageServiceProperties;
+import com.azure.storage.queue.models.StorageServiceStats;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Queue service async client
+ */
+public final class QueueServiceAsyncClient {
+    private final AzureQueueStorageImpl client;
+
+    /**
+     * Constructor used by the builder
+     * @param endpoint URL of the storage account queue endpoint
+     * @param httpPipeline Http pipeline
+     */
+    QueueServiceAsyncClient(URL endpoint, HttpPipeline httpPipeline) {
+        this.client = new AzureQueueStorageImpl(httpPipeline).withUrl(endpoint.toString());
+    }
+
+    /**
+     * @return a new client builder instance
+     */
+    public static QueueServiceAsyncClientBuilder builder() {
+        return new QueueServiceAsyncClientBuilder();
+    }
+
+    /**
+     * @return URL of the storage account queue endpoint
+     */
+    public String url() {
+        return client.url();
+    }
+
+    /**
+     * Gets a QueueAsyncClient that is targeting the specified queue
+     * @param queueName Name of the queue
+     * @return QueueAsyncClient that interacts with the specified queue
+     */
+    public QueueAsyncClient getQueueAsyncClient(String queueName) {
+        return new QueueAsyncClient(client, queueName);
+    }
+
+    /**
+     * Creates a new queue and returns the client to interact with it
+     * @param queueName Name of the queue
+     * @return the client to interact with the new queue
+     */
+    public QueueAsyncClient createQueue(String queueName) {
```

Comment:
Interesting - you have 'createQueue' here but not previously.

Reply:
We were trying to have the QueueServiceClient make it obvious that it was creating a queue resource